src/app/dashboard/train/components/NewModel.tsx

```diff
@@ -0,0 +1,314 @@
+import axios from "axios";
+
+interface IProps {
+    setPage: (page: number | null) => void
+    setModal: (modal: number | null) => void
+    setTrain: (train: boolean) => void
+    train: boolean
+}
+
+
+
+const NewModel = ({ setPage, setModal, setTrain, train }: IProps) => {
+    const trainScriptInput = useRef<HTMLInputElement | null>(null)
+    const requirementInput = useRef<HTMLInputElement | null>(null)
+    const dataSetInput = useRef<HTMLInputElement | null>(null)
+
+    const [trainScript, setTrainScript] = useState<File | null>(null)
+    const [requirementsScript, setRequirementsScript] = useState<File | null>(null)
+    const [modelName, setModelName] = useState<string>("")
+    const [dataSet, setDataSet] = useState<File[]>([])
+
+
+
+    //truncate file name
+    const truncateFileName = (
+        fileName: string,
+        size = 10
+    ) => {
+        const split = fileName.split('.');
+        const filename = split[0];
+        const extension = split[1];
+        if (filename.length > size) {
+            let currFileName = filename.substring(0, size);
+            return currFileName + '....' + extension;
+        } else {
+            return fileName
+        }
+
+    };
+
+    // regex for file extension
+    const regex = new RegExp("[^.]+$");
+
+    //format file size
+    const formatBytes = (bytes: number, decimals = 2) => {
+        if (!+bytes) return '0 Bytes'
+
+        const k = 1024
+        const dm = decimals < 0 ? 0 : decimals
+        const sizes = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
+
+        const i = Math.floor(Math.log(bytes) / Math.log(k))
+
+        return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`
+    }
+
+    //train script
+    const handleTrainScriptDragOver = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+
+    }
+
+    const handleTrainScriptDrop = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+        const files = e.dataTransfer.files
+        if (!files || files.length > 1) {
+            console.log("upload one file")
+            return
+        }
+
+        if (files[0].name.match(regex)?.[0] === "ipynb") {
+            //store file
+            setTrainScript(files[0])
+        } else {
+            console.log("upload a note book")
+        }
+
+    }
+
+    //requirements.txt
+    const handleRequirementsScriptDragOver = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+
+    }
+
+
+    const handleRequirementsScriptDrop = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+        const files = e.dataTransfer.files
+        if (!files || files.length > 1) {
+            console.log("upload one file")
+            return
+        }
+        console.log(files)
+
+        if (files[0].name.toUpperCase() === "REQUIREMENTS.TXT") {
+            //store file
+            setRequirementsScript(files[0])
+        } else {
+            console.log("upload a requirements.txt file")
```

Comment:
no necessary, lets forget about requirements.txt for now. Please comment or disable for now